lib/cloudant.js

```diff
@@ -862,19 +884,25 @@ Cloudant.prototype.replaceById = function replaceById(
   var self = this;
   var mo = self.selectModel(model);
   var idName = self.idName(model);
-  data[idName] = id;
+  data[idName] = id.toString();
 
-  self.getCurrentRevision(model, id, function(err, rev) {
+  var replaceHandler = function(err, id) {
     if (err) return cb(err);
-    if (rev) data._rev = rev;
-    self._insert(model, data, function(err, id) {
+    mo.db.get(id, function(err, doc) {
       if (err) return cb(err);
-      mo.db.get(id, function(err, doc) {
-        if (err) return cb(err);
-        cb(null, self.fromDB(model, mo, doc));
-      });
+      cb(null, self.fromDB(model, mo, doc));
     });
-  });
+  };
+
+  if (!data._rev) {
+    self.getCurrentRevision(model, id, function(err, rev) {
```

Comment:
@kjdelisle `replaceById` apparently does not store the `_rev` value like other `replaceOrCreate` or `updateOrCreate` method. We might have to just keep the `getCurrentRevision` or fix the juggler dao.